### PR TITLE
do not allow update with empty document

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1616/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1617/
Test PASSed.

**Comment by rogpeppe** [comment]
LGTM, thanks.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1618/
Test PASSed.

File: internal/charmstore/store_test.go [hunk 1]
```diff
@@ -1723,20 +1723,46 @@ func (s *StoreSuite) TestFindBestEntityWithMultiSeriesCharms(c *gc.C) {
...
+	},
+}, {
+	about:  "nil update",
+	url:    "~charmers/trusty/wordpress-10",
+	update: nil,
+	check: func(c *gc.C, e *mongodoc.Entity) {
+		c.Assert(e.PromulgatedURL, gc.Not(gc.IsNil))
+		c.Assert(e.PromulgatedURL.String(), gc.Equals, "cs:trusty/wordpress-4")
+	},
 }}
```
Comment (fabricematrat) [inline internal/charmstore/store_test.go:1759]: Do we have a test where we pass a bson with data which are unknown to charmstore ?

**Comment by fabricematrat** [comment]
:+1:

File: internal/charmstore/store_test.go (hunk 1 above)
Comment (mhilton) [inline internal/charmstore/store_test.go:1759]: No, but that will succeed anyway, and that's probably fine for this case.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**